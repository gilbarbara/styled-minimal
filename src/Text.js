import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { borders, display, fontFamily, fontSize, fontWeight, space } from 'styled-system';

export const StyledText = styled.span`
  ${borders};
  ${display};
  ${fontFamily};
  ${fontSize};
  ${fontWeight};
  ${space};
`;

const Text = ({ children, ...props }) => (
  <StyledText {...props}>{children}</StyledText>
);

Text.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Text;
