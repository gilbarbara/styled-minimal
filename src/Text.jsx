import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { withComponent } from './utils/helpers';
import { TextStyles } from './utils/system';

export const StyledText = styled.span`
  ${TextStyles.base};
`;

StyledText.propTypes = {
  children: PropTypes.node.isRequired,
};

export default withComponent(StyledText, 'Text');
