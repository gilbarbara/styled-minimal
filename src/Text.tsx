import * as React from 'react';
import styled, { css } from 'styled-components';

import { baseStyles, lineHeightStyles, sx, system } from './utils/system';

import { TextProps } from './types';

export const StyledText = styled('span')<TextProps>(
  props => css`
    ${baseStyles};
    color: ${props.muted ? '#999' : 'inherit'};
    line-height: ${lineHeightStyles};
  `,
  sx,
  system,
);

export const Text = React.forwardRef<HTMLSpanElement, TextProps>((props, ref) => (
  <StyledText ref={ref} {...props} />
));
