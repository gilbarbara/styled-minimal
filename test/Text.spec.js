import React from 'react';
import { mount } from 'enzyme';

import Text from '../src/Text';

describe('Text', () => {
  const wrapper = mount(
    <React.Fragment>
      <Text>Far far away, behind the word mountains, </Text>
      <Text>far from the countries Vokalia and Consonantia, there live the blind texts.</Text>
    </React.Fragment>
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
