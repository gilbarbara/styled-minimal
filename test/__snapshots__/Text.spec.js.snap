// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render properly 1`] = `
Array [
  .c0 {
  box-sizing: border-box;
}

<Text
    as="span"
  >
    <span
      className="c0"
    >
      Far far away, behind the word mountains, 
    </span>
  </Text>,
  .c0 {
  box-sizing: border-box;
}

<Text
    as="span"
  >
    <span
      className="c0"
    >
      far from the countries Vokalia and Consonantia, there live the blind texts.
    </span>
  </Text>,
]
`;
